sdk/keyvault/keyvault-keys/README.md

```diff
@@ -40,6 +40,18 @@ To quickly create the needed Key Vault resources in Azure and to receive a conne
 
 [![](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-sdk-for-js%2Fmaster%2Fsdk%2Fkeyvault%2Ftest-resources.json)
 
+To fill this form:
+
+1. If you know about ARM templates, or you're interested to learn about them, we recommend you to take some time to read through the [ARM template used by the deploy button in this readme](https://github.com/Azure/azure-sdk-for-js/blob/keyvault/deploy-button-guide/sdk/keyvault/test-resources.json). You can read more about ARM templates in the [Azure Resource Manager templates overview](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview).
+2. The subscription, resource group and location are required fields, they help specify where to create the KeyVault.
+3. On the "Base Name", you'll see `[resourceGroup().name]`. Remove the contents of this field and type any alphanumeric string. This string will be used to generate unique names for the resources that will be created.
+4. To fill the "Tenant Id" field, copy it from the one you'll be able to see once you go to the [overview page of the Azure Active Directory section of the Azure portal](https://ms.portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/Overview).
+5. To fill the "Test Application Oid", you'll first need to create an Azure Active Directory application (Read: [How to: Use the portal to create an Azure AD application and service principal that can access resources](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)). Once created, go to your application's overview page, and you'll see an "Object ID" property. This is the one that you want to copy and paste on the form.
+6. Finally, accept the terms and conditions, then click on the "Purchase" button.
+7. A badge will appear on the top right of the portal, indicating that the resource is being created. You can go to your resource group's page, then go to the "Deployments" section to see a history of all the deployments you've made. You'll see that those deployments that have used an ARM template will say in their names "Microsoft.Template". Once you click in one of them, you'll see at the left "Outputs" as one of the menu options. If your deployment has succeeded, you will see the name of the KeyVault that was just created.
```

Comment:
```suggestion
7. A notification will appear on the top right of the portal indicating that the resource is being created. Clicking on the notification will show you deployment status. You will also find this information under your resource group's *Deployments* page.
```

* I think portal calls these things notifications not badges...
* General tutorial about managing resources seems out of scope for this document.